src/greeter/UserModel.cpp

```diff
@@ -134,16 +134,20 @@ namespace SDDM {
                 d->lastIndex = i;
 
             if (avatarsEnabled) {
-                const QString userFace = QStringLiteral("%1/.face.icon").arg(user->homeDir);
                 const QString systemFace = QStringLiteral("%1/%2.face.icon").arg(facesDir).arg(user->name);
-                QString accountsServiceFace = QStringLiteral("/var/lib/AccountsService/icons/%1").arg(user->name);
-
-                if (QFile::exists(userFace))
-                    user->icon = QStringLiteral("file://%1").arg(userFace);
-                else if (QFile::exists(accountsServiceFace))
-                    user->icon = accountsServiceFace;
-                else if (QFile::exists(systemFace))
+                 
+                if (QFile::exists(systemFace))  {
                     user->icon = QStringLiteral("file://%1").arg(systemFace);
+                } else {
+                    const QString userFace = QStringLiteral("%1/.face.icon").arg(user->homeDir);
+                    QString accountsServiceFace = QStringLiteral("/var/lib/AccountsService/icons/%1").arg(user->name);
+
+                    if (QFile::exists(userFace)) {
+                        user->icon = QStringLiteral("file://%1").arg(userFace);
+                    } else if (QFile::exists(accountsServiceFace)) {
+                        user->icon = accountsServiceFace;
```

Comment:
the diff could be more minimal, only bringing up two lines, instead of changing the structure to have two different imbricated `if`

Reply:
I pushed new code following your advice